src/freeze/thaw.rs

```diff
@@ -0,0 +1,395 @@
+    let candy_machine_id = match args.candy_machine {
+        Some(ref candy_machine_id) => candy_machine_id.to_owned(),
+        None => {
+            let cache = load_cache(&args.cache, false)?;
+            cache.program.candy_machine
+        }
+    };
+
+    let candy_pubkey = Pubkey::from_str(&candy_machine_id)
+        .map_err(|_| anyhow!("Failed to parse candy machine id: {}", &candy_machine_id))?;
+
+    let total_steps = if args.all { 4 } else { 2 };
+
+    println!(
+        "{} {}Loading candy machine",
+        style(format!("[1/{}]", total_steps)).bold().dim(),
+        LOOKING_GLASS_EMOJI
+    );
+    println!("{} {}", style("Candy machine ID:").bold(), candy_machine_id);
+
+    let pb = spinner_with_style();
+    pb.set_message("Connecting...");
+    let _candy_machine_state =
+        get_candy_machine_state(&sugar_config, &Pubkey::from_str(&candy_machine_id)?)?;
+
+    pb.finish_with_message("Done");
+
+    if !args.all {
+        println!(
+            "\n{} {}Thawing single NFT. . .",
+            style(format!("[2/{}]", total_steps)).bold().dim(),
+            MONEY_BAG_EMOJI
+        );
+
+        let nft_mint = if let Some(nft_mint) = &args.nft_mint {
+            nft_mint.to_owned()
+        } else {
+            return Err(anyhow!("NFT mint is required if thawing a single NFT"));
+        };
+
+        let nft_mint_pubkey = Pubkey::from_str(&nft_mint)
+            .map_err(|_| anyhow!("Failed to parse nft mint id: {}", &nft_mint))?;
+
+        let config = Arc::new(sugar_config);
+
+        let request = RpcRequest::Custom {
+            method: "getTokenLargestAccounts",
+        };
+        let params = json!([nft_mint, { "commitment": "confirmed" }]);
+        let result: JRpcResponse = rpc_client.send(request, params).unwrap();
+
+        let token_accounts: Vec<TokenAccount> = result
+            .value
+            .into_iter()
+            .filter(|account| account.amount.parse::<u64>().unwrap() == 1)
+            .collect();
+
+        if token_accounts.len() > 1 {
+            return Err(anyhow!(
+                "Mint account {} had more than one token account with 1 token",
+                nft_mint
+            ));
+        }
+
+        if token_accounts.is_empty() {
+            return Err(anyhow!(
+                "Mint account {} had zero token accounts with 1 token",
+                nft_mint
+            ));
+        }
+
+        let token_account = Pubkey::from_str(&token_accounts[0].address).unwrap();
+
+        let account = program
+            .rpc()
+            .get_account_with_commitment(&token_account, CommitmentConfig::confirmed())
+            .unwrap()
+            .value
+            .unwrap();
+        let account_data = SplAccount::unpack(&account.data).unwrap();
+        let owner = account_data.owner;
+
+        // Only thaw frozen accounts.
+        if !account_data.is_frozen() {
+            println!("\n NFT is already thawed! NFT");
+            std::process::exit(0);
```

Comment:
Change this line for `return Ok(());` so the command exits gracefully.

Reply:
[Fixed.](https://github.com/metaplex-foundation/sugar/pull/345/commits/274e3759ac42808a82f4e05dd5a0fa7ab591961f)